ml-prims/src/cuda_utils.h

```diff
@@ -185,6 +185,45 @@ DI void myAtomicReduce(long long *address, long long val, ReduceLambda op) {
   } while (assumed != old);
 }
 
+
+/**
+ * @brief Provide atomic min operation.
+ * @tparam T: data type for input data (float or double).
+ * @param[in] address: address to read old value from, and to atomically update w/ min(old value, val)
+ * @param[in] val: new value to compare with old
+ */
+template<typename T>
+DI T myAtomicMin(T *address, T val);
+
+/**
+ * @brief Provide atomic max operation.
+ * @tparam T: data type for input data (float or double).
+ * @param[in] address: address to read old value from, and to atomically update w/ max(old value, val)
+ * @param[in] val: new value to compare with old
+ */
+template<typename T>
+DI T myAtomicMax(T *address, T val);
+
+DI float myAtomicMin(float *address, float val) {
```

Comment:
There is a better way to do this than `atomicCAS()` in a loop:
- for values >= 0, just use `atomicMin()` for int
- for the general case, additional bit flipping is required both during the atomic operation, and for the final result (after all atomics)

This is also applicable to `atomicMin()` and double.

Reply:
This is certainly a faster way of doing atomicMin/Max. Thanks @canonizer .

IMO, such bit twiddling operations are better handled in a separate PR. I have filed issue #494 so that this is not missed out.